tests/lib/prepare.sh

```diff
@@ -192,6 +192,54 @@ update_core_snap_for_classic_reexec() {
     done
 }
 
+prepare_memory_limit_override() {
+    # set up memory limits for snapd bu default unless explicit requested not to
+    # or the system is known to be problematic
+    local set_limit=1
+
+    case "$SPREAD_SYSTEM" in
+        ubuntu-core-16-*|ubuntu-core-18-*|ubuntu-16.04-*)
+            # the tests on UC16 and UC18 have demonstrated that the memory limit
+            # state claimed by systemd may be out of sync with actual memory
+            # controller setting for the snapd.service cgroup
+            set_limit=0
+            ;;
+        amazon-linux-*)
```

Comment:
I would have guessed Centos 7 is in the same boat as amazon-linux ?

Reply:
actually the tests did not fail on centos 7, but I had to workaround some weird inconsistency on 16.04 and 18.04. Perhaps those are real bugs in systemd that got patched in rhel.